File: CS/E5198/CustomFontConverter.cs
using DevExpress.Utils.Serializing;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Web;

namespace E5198 {
    public class CustomFontConverter : PrintingSystemXmlSerializer.FontConverter {
        protected override object CreateObject(string[] values) {
            Font font = base.CreateObject(values) as Font;
            if (values.Length > 0) {
                FontFamily fontFamily = CustomFontsHelper.GetFamily(values[0]);
                if (fontFamily != null)
                    return new Font(fontFamily, font.Size, font.Style, font.Unit, font.GdiCharSet, font.GdiVerticalFont);
            }
            return font;
        }
    }
}

File: VB/E5198/CustomFontConverter.vb
Imports DevExpress.Utils.Serializing
Imports System.Drawing

Namespace E5198

    Public Class CustomFontConverter
        Inherits PrintingSystemXmlSerializer.FontConverter

        Protected Overrides Function CreateObject(ByVal values As String()) As Object
            Dim font As Font = TryCast(MyBase.CreateObject(values), Font)
            If values.Length > 0 Then
                Dim fontFamily As FontFamily = GetFamily(values(0))
                If fontFamily IsNot Nothing Then Return New Font(fontFamily, font.Size, font.Style, font.Unit, font.GdiCharSet, font.GdiVerticalFont)
            End If

            Return font
        End Function
    End Class
End Namespace
